app/src/main/java/dev/sasikanth/pinnit/ShortcutReceiverActivity.kt

```diff
@@ -10,10 +10,14 @@ import dev.sasikanth.pinnit.editor.EditorTransition.SharedAxis
 
 class ShortcutReceiverActivity : AppCompatActivity() {
 
+  companion object {
+    const val MESSAGE = "message"
+  }
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
 
-    val receivedText = intent.getStringExtra(EXTRA_TEXT)
+    val receivedText = intent.getStringExtra(EXTRA_TEXT) ?: intent?.data?.getQueryParameter(MESSAGE)
```

Comment:
Same comment as above, I think we should use a separate activity for handling app actions. Because if we were to add a new app shortcut or handle shared text differently, we have to remember to handle the app action data. So it's better to separate these out. WDYT?